### PR TITLE
Story7 task

### PR DESCRIPTION
**Pull request by Henry-Tottle**

### CODE REVIEW

File: product.php [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
 use FurnitureStoreApi\Services\UnitConversionService;
-try
-{
-    if ($_GET['id']>ProductHydrator::getMaxProducts())
-    {
+use FurnitureStoreApi\Services\CurrencyConversionClass;
+use FurnitureStoreApi\Exceptions\InvalidCurrencyException;
+
+try {
+    if ($_GET['id'] > ProductHydrator::getMaxProducts()) {
```
Comment (matt-ioa) [inline product.php]: Didn't you also have an `is_numeric` check on this as well?

File: products.php [hunk 2]
```diff
@@ -1,34 +1,35 @@
...
+use FurnitureStoreApi\Services\CurrencyConversionClass;
+
 HeaderService::setHeaders();
 
-try
-{
-    if ($_GET['cat']>CategoryHydrator::getMaxCategory()||$_GET['cat']<0||!is_numeric($_GET['cat']))
-    {
+try {
+    if ($_GET['cat'] > CategoryHydrator::getMaxCategory() || $_GET['cat'] < 0 || !is_numeric($_GET['cat'])) {
```
Comment (matt-ioa) [inline products.php:15]: A bit counterintuitive that this is actually the opposite (NOT) case to the way you were checking the id before, you can see how this would be confusing.

File: src/Products/SimpleProductEntity.php [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+use FurnitureStoreApi\Services\CurrencyConversionClass;
 use FurnitureStoreApi\Services\HeaderService;
 HeaderService::setHeaders();
 
 class SimpleProductEntity implements \JsonSerializable
 {
     protected int $id;
     protected string $name;
-    protected float $price;
+    protected string $price;
```
Comment (matt-ioa) [inline src/Products/SimpleProductEntity.php]: Why did you change this to a string?

File: src/Services/CurrencyConversionClass.php [hunk 4]
```diff
@@ -1,26 +1,41 @@
...
+        {
+            self::$currency = $currency;
+        }
+        else
+        {
+            throw new InvalidCurrencyException();
+
+        }
+    }
+    public static function currencyConverter(string $price): string
```
Comment (matt-ioa) [inline src/Services/CurrencyConversionClass.php]: It seems odd to be that this would take a currency amount as a string and return a string. Return a string, maybe, but the input should be a float, really.

**Review by matt-ioa — CHANGES_REQUESTED** [review]
A couple of changes requested, take a look :eyes:

File: src/Products/SimpleProductEntity.php (hunk 3 above)
Comment (Henry-Tottle) [reply src/Products/SimpleProductEntity.php]: That was feedback from a different Story, but happy to change it

File: products.php (hunk 2 above)
Comment (Henry-Tottle) [reply products.php:15]: From my understanding of this, this is checking that if 'cat' is greater than max, or less than 0, or is not numeric, it throws an error.  So I have changed the other one to match.  But maybe that isn't the solution?

File: src/Products/SimpleProductEntity.php (hunk 3 above)
Comment (Henry-Tottle) [reply src/Products/SimpleProductEntity.php]: I think I get it.

File: src/Services/CurrencyConversionClass.php (hunk 4 above)
Comment (Henry-Tottle) [reply src/Services/CurrencyConversionClass.php]: This now takes a float and returns a string, because the number formatting function we use returns a string and that is the only way we can get certain numbers to display as 2 decimal points without rounding up to one decimal point.

**Comment by Henry-Tottle** [comment]
I think I've made the right changes in what I understand.

**Review by matt-ioa — APPROVED** [review]
Great work, well done!
